From 36a34f886eceffa6ca6b79522ba7a236ec7444f2 Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Wed, 3 Dec 2014 07:51:12 +0100
Subject: [PATCH] wifi: Allow building without Wifi HAL

 * For example devices with older broadcom chips with driver built
   as module doesn't like this. When the device goes into suspend
   mode, the wifi service crashes and the device soft reboots.

 * To not build the HAL, use:
   BOARD_NO_WIFI_HAL := true

Change-Id: Ia186ce3a59a082cfe98d283492fa450014af52f4

Conflicts:
	service/Android.mk
---
 service/Android.mk | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

project frameworks/opt/net/wifi/
diff --git a/frameworks/opt/net/wifi/service/Android.mk b/frameworks/opt/net/wifi/service/Android.mk
index 08b9ff0..c17553c 100755
--- a/frameworks/opt/net/wifi/service/Android.mk
+++ b/frameworks/opt/net/wifi/service/Android.mk
@@ -42,14 +42,16 @@ include $(BUILD_STATIC_LIBRARY)
 # ============================================================
 LIB_WIFI_HAL := libwifi-hal
 
-ifeq ($(BOARD_WLAN_DEVICE), bcmdhd)
-  LIB_WIFI_HAL := libwifi-hal-bcm
-else ifeq ($(BOARD_WLAN_DEVICE), qcwcn)
-  LIB_WIFI_HAL := libwifi-hal-qcom
-else ifeq ($(BOARD_WLAN_DEVICE), mrvl)
-  # this is commented because none of the nexus devices
-  # that sport Marvell's wifi have support for HAL
-  # LIB_WIFI_HAL := libwifi-hal-mrvl
+ifneq ($(BOARD_NO_WIFI_HAL), true)
+  ifeq ($(BOARD_WLAN_DEVICE), bcmdhd)
+    LIB_WIFI_HAL := libwifi-hal-bcm
+  else ifeq ($(BOARD_WLAN_DEVICE), qcwcn)
+    LIB_WIFI_HAL := libwifi-hal-qcom
+  else ifeq ($(BOARD_WLAN_DEVICE), mrvl)
+    # this is commented because none of the nexus devices
+    # that sport Marvell's wifi have support for HAL
+    # LIB_WIFI_HAL := libwifi-hal-mrvl
+  endif
 else ifeq ($(BOARD_WLAN_DEVICE), MediaTek)
   # support MTK WIFI HAL
   LIB_WIFI_HAL := libwifi-hal-mt66xx
-- 
1.9.1

